src/ts/controller/validator.ts

```diff
@@ -0,0 +1,51 @@
+import { REGEXP, EQUAL } from "../constants/index.js";
+
+export const isNumber = (str: string): boolean => {
+  return /^[\d.]+(?:e-?\d+)?$/.test(str);
+};
+
+export const checkFirstTotalValue = (
+  totalValue: string,
+  clickValue: string
+): boolean => {
+  return totalValue === "0" && (isNumber(clickValue) || clickValue === "-");
+};
+
+export const checkPreventClickValue = (
+  totalValue: string,
+  clickValue: string
+): boolean => {
+  const tmpValue = totalValue + clickValue;
+  const tmpValueArray: RegExpMatchArray | null = tmpValue.match(
+    "(" +
+      REGEXP.NUMBERS +
+      ")?(" +
+      REGEXP.OPERATORS +
+      ")?(" +
+      REGEXP.NUMBERS +
+      ")?"
+  );
+  if (tmpValueArray) {
+    if (!tmpValueArray[2] && tmpValueArray[3]) {
+      return true;
+    }
+    if (
+      tmpValueArray[2] &&
+      !isNumber(totalValue.charAt(totalValue.length - 1)) &&
+      !isNumber(clickValue)
+    ) {
+      return true;
+    }
+    if (tmpValueArray[3] && !isNumber(clickValue) && clickValue !== EQUAL) {
+      return true;
+    }
+    if (
+      tmpValueArray[3] &&
+      tmpValue.length !== tmpValueArray[0].length &&
+      clickValue !== EQUAL
+    ) {
+      return true;
+    }
```

Comment:
모든 if 문이 안에서 return 하고 있으니까 이렇게 바꾸면 좋겠네요 ! 

```suggestion
    if (!tmpValueArray[2] && tmpValueArray[3]) {
      return true;
    }
    else if (
      tmpValueArray[2] &&
      !isNumber(totalValue.charAt(totalValue.length - 1)) &&
      !isNumber(clickValue)
    ) {
      return true;
    }
    else if (tmpValueArray[3] && !isNumber(clickValue) && clickValue !== EQUAL) {
      return true;
    }
    else if (
      tmpValueArray[3] &&
      tmpValue.length !== tmpValueArray[0].length &&
      clickValue !== EQUAL
    ) {
      return true;
    }
```

Reply:
그러네요. 순차적인 처리라 그게 맞는 듯 합니다.